hazelcast/src/main/java/com/hazelcast/concurrent/lock/WaitersInfo.java

```diff
@@ -29,8 +29,10 @@
 
 final class WaitersInfo implements IdentifiedDataSerializable {
 
+    private static final int INITAIL_WAITER_SIZE = 4;
+
     private String conditionId;
-    private Set<ConditionWaiter> waiters = new HashSet<ConditionWaiter>(2);
+    private Set<ConditionWaiter> waiters = new HashSet<ConditionWaiter>(INITAIL_WAITER_SIZE);
```

Comment:
The value has changed from 2 to 4, is this intentional?

Reply:
I will check that. Actually I wanted to go through the whole PR once more as I initially based it on a community PR. The guy put in an effort to understand parts of the code and try and guess the size on each change.

Reply:
BTW, there is a typo in the constant name :) `INITAIL_WAITER_SIZE` should be `INITIAL_WAITER_SIZE` :)

Reply:
Changing back to 2 and using the `com.hazelcast.util.SetUtil#createHashSet` method.